src/main/java/org/opensrp/web/rest/StockResource.java

```diff
@@ -113,7 +113,9 @@ public Stock getByUniqueId(String uniqueId) {
 				lastSyncedServerVersion = Long.parseLong(serverVersion);
 			}
 			List<Stock> stocks = new ArrayList<Stock>();
-			stocks = stockService.findAllStocks(lastSyncedServerVersion,limit);
+			StockSearchBean stockSearchBean = new StockSearchBean();
+			stockSearchBean.setServerVersion(lastSyncedServerVersion);
+			stocks = stockService.findStocks(stockSearchBean, BaseEntity.SERVER_VERSIOIN, "asc", limit);
```

Comment:
Using this will do an inner join on `core.stock` and the `core.product_catalogue`, this will work okay for eusm but other projects dont have productCatalogues.